src/ctap2/attestation.rs

```diff
@@ -1078,6 +1079,24 @@ pub mod test {
         );
     }
 
+    #[test]
+    fn test_empty_extension_data() {
+        let mut parsed_auth_data: AuthenticatorData = from_slice(&SAMPLE_AUTH_DATA_MAKE_CREDENTIAL).unwrap();
+        assert!(parsed_auth_data.flags.contains(AuthenticatorDataFlags::EXTENSION_DATA));
+
+        // Remove the extension data but keep the extension data flag set.
+        parsed_auth_data.extensions = Default::default();
+        let with_flag = to_vec(&parsed_auth_data).expect("could not serialize auth data");
+        // The serialized auth data should end with an empty map (CBOR 0xA0).
+        assert_eq!(with_flag[with_flag.len()-1], 0xA0);
```

Comment:
super tiny nit: I would expect this to be formatted as `with_flag[with_flag.len() - 1]` - does `cargo fmt` not format tests by default?

Reply:
whoops, I didn't run rustfmt and our CI doesn't enforce it.